dataextractor.py

```diff
@@ -0,0 +1,300 @@
+#Importing
+from rdkit import Chem, RDConfig, Geometry
+from rdkit.Chem import AllChem, TorsionFingerprints
+import os, sys
+import openeye.oechem as oechem
+from openeye import oeshape
+import pandas as pd
+import numpy as np
+import itertools
+
+
+def tanimotocombo(ref_mol, query_mol):
+    """This is the TanimotoCombo function. It takes in two OEMols. 
+    It does not matter (for our purposes) which mol is the reference mol and 
+    which one is the query mol. The TanimotoCombo distance between the two
+    mols is the same no matter which is the reference and which is the query.
+    The GetTanimotoCombo() function comes from openeye. 
+    """
+    # Prepare reference molecule for calculation
+    # With default options this will remove any explicit
+    # hydrogens present and add color atoms
+    prep = oeshape.OEOverlapPrep()
+    prep.Prep(ref_mol)
+    # Get appropriate function to calculate both shape and color
+    # By default the OEOverlapFunc contains OEGridShapeFunc for shape
+    # and OEExactColorFunc for color
+    func = oeshape.OEOverlapFunc()
+    func.SetupRef(ref_mol)
+    res = oeshape.OEOverlapResults()
+    prep.Prep(query_mol)
+    func.Overlap(query_mol, res)
+    return("%s" % res.GetTanimotoCombo())
+
+
+def TFDr(ref_mol, query_mol):
+    """
+    This is the TFDr script. 
+    It makes use of RDKit's TFD calculation and the function rdmol_from_oemol.
+    TFDr takes in two OEMOLs. 
+    It does not matter which mol is the ref mol and which is the querymol. 
+    TFD metric is the same no matter which is the ref and which is the query.
+    First, OEmols are made RDKit compatible. Then,
+    TFD is computed and returned using RDKit's TorsionFingerprints
+    Module. Takes one input reference mol2 and one input query mol2.
+    """
+    #converts refmol to one readable by RDKit
+    rrdmol2 = rdmol_from_oemol(ref_mol)
+    #converts querymol to one readable by RDKit
+    qrdmol2 = rdmol_from_oemol(query_mol)
+    #If there was a mistake in the conversion process, return -1
+    if (Chem.MolToSmiles(qrdmol2) != Chem.MolToSmiles(rrdmol2)):
+        tfd = -1
+    else:
+        #calculates the TFD
+        tfd = TorsionFingerprints.GetTFDBetweenMolecules(rrdmol2, qrdmol2)
+    return tfd
+
+
+
+def rdmol_from_oemol(oemol):
```

Comment:
should write a check to make sure the input oemol is not already RDKit-compatible

Reply:
Re this comment: Create an Issue on the repo to track that you need to do this. Don't bury "to do" items in comments on GitHub as the comments become hard to find once we change the code or merge the pull request.